### PR TITLE
KFSTP-1697

### PR DESCRIPTION
**Pull request by smith750**
Add logic to force batch-created CINVs which would do not pass routing validation to route to funds manager.

### CODE REVIEW

File: work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java [hunk 1]
```diff
@@ -463,12 +463,13 @@ public boolean answerSplitNodeQuestion(String nodeName) throws UnsupportedOperat
...
-     * @return
+     * @return true if this CINV should route to the fund managers, false if it should skip
      */
     private boolean isRequiresFundingManagerApproval() {
-        // if auto approve on the award is false or suspension exists, then we need to have funds manager approve.
+        final ContractsGrantsInvoiceDocumentService contractsGrantsInvoiceDocumentService = SpringContext.getBean(ContractsGrantsInvoiceDocumentService.class);
+        // if auto approve on the award is false or suspension exists or the award is auto-approve but fails to pass validation, then we need to have funds manager approve.
         boolean result;
-        result = !getInvoiceGeneralDetail().getAward().getAutoApproveIndicator() || !this.getInvoiceSuspensionCategories().isEmpty();
+        result =  !getInvoiceSuspensionCategories().isEmpty() || !getInvoiceGeneralDetail().getAward().getAutoApproveIndicator() || (contractsGrantsInvoiceDocumentService.isDocumentBatchCreated(this) && !contractsGrantsInvoiceDocumentService.doesInvoicePassValidation(this));
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java]: I suppose we could use CollectionUtils.isNotEmpty() here but I'm not sure that buys us much.

File: work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java [hunk 2]
```diff
@@ -1891,6 +1901,60 @@ public void recalculateSourceAccountingLineTotals(ContractsGrantsInvoiceDocument
...
+            return StringUtils.equalsIgnoreCase(document.getFinancialSystemDocumentHeader().getInitiatorPrincipalId(), batchJobInitiatorPrincipal.getPrincipalId());
+        }
+        return false;
+    }
+
+    /**
+     * @see org.kuali.kfs.module.ar.document.service.ContractsGrantsInvoiceDocumentService#doesInvoicePassValidation(org.kuali.kfs.module.ar.document.ContractsGrantsInvoiceDocument)
+     */
+    @Override
+    public boolean doesInvoicePassValidation(final ContractsGrantsInvoiceDocument document) {
```
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java]: This is an interesting method. I wish a) we could call the validation more directly, but that's probably not possible and b) we didn't have to throw and catch generic Exception, but that's probably not avoidable.

File: work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java [hunk 3]
```diff
@@ -463,12 +464,13 @@ public boolean answerSplitNodeQuestion(String nodeName) throws UnsupportedOperat
...
-     * @return
+     * @return true if this CINV should route to the fund managers, false if it should skip
      */
     private boolean isRequiresFundingManagerApproval() {
-        // if auto approve on the award is false or suspension exists, then we need to have funds manager approve.
+        final ContractsGrantsInvoiceDocumentService contractsGrantsInvoiceDocumentService = SpringContext.getBean(ContractsGrantsInvoiceDocumentService.class);
+        // if auto approve on the award is false or suspension exists or the award is auto-approve but fails to pass validation, then we need to have funds manager approve.
         boolean result;
-        result = !getInvoiceGeneralDetail().getAward().getAutoApproveIndicator() || !this.getInvoiceSuspensionCategories().isEmpty();
+        result =  !CollectionUtils.isEmpty(getInvoiceSuspensionCategories()) || !getInvoiceGeneralDetail().getAward().getAutoApproveIndicator() || (contractsGrantsInvoiceDocumentService.isDocumentBatchCreated(this) && !contractsGrantsInvoiceDocumentService.doesInvoicePassValidation(this));
```
Comment (smith750) [inline work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java]: Bryan - can't hurt.  I've made the change & I'm testing it real quick.

File: work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java (hunk 2 above)
Comment (smith750) [inline work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java]: a) My concern with calling the validation more directly is simply that the institution may have changed something.  I can't get any closer to the actual validation before I start running into that issue.  b) Yeah, Exception is handed to me straight from java.util.Concurrent.Callable; no way to change.  Thoughts?

Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java]: Yeah, that's kinda what I figured on both counts. So I think it stays as is, thanks.

File: work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java (hunk 3 above)
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java]: thanks

**Pull request merged**